blog/buildFrontMatterCache.ts

```diff
@@ -30,31 +31,25 @@ function reverseSort(a: PostFrontMatterWithSlug, b: PostFrontMatterWithSlug) {
   return bDate.getTime() - aDate.getTime();
 }
 
-function validateFrontMatter(frontMatter: PostFrontMatter): never | void {
-  const hasTitle = !!frontMatter.title;
-  const hasAuthor = !!frontMatter.author;
-  const hasDate = !!frontMatter.date;
-  const isValid = hasTitle && hasAuthor && hasDate;
-  let missingFields: string[] = [];
-  if (!isValid) {
-    !hasTitle && missingFields.push("title");
-    !hasAuthor && missingFields.push("author");
-    !hasDate && missingFields.push("date");
-    throw new Error(
-      `${
-        hasTitle ? `"${frontMatter.title}"` : "A blog post"
-      } was missing this required front matter: [${missingFields.join(", ")}].`
-    );
-  }
+function validateFrontMatter(
+  frontMatter: Partial<PostFrontMatter>
+): frontMatter is PostFrontMatter {
+  const title = frontMatter.title;
+  const author = frontMatter.author;
+  const date = frontMatter.date;
+  invariant(title, "A blog post was missing a title.");
+  invariant(author, `Blog post ${title} was missing an author.`);
+  invariant(date, `Blog post ${title} was missing an date.`);
+  return true;
 }
 
 /**
  * Builds the front matter of all posts into a `PostFrontMatterCollection`
  * as a JSON string.
  */
-export function buildFrontMatter(): string {
+export function buildFrontMatterCache(): void {
```

Comment:
this is nitpicky, but take a look at some of these method names. for instance, a more accurate name for this might be `writeFrontMatterCache`. and the `parseMdx` might be more accurate as `bundleMdx`. 

naming is hard but now that you're coalescing around the end of this pr, take a quick sweep through on the names. (but i'm also not going to block this pr off of naming.)

Reply:
I've changed the names a lot already lol. But it seems like a good name for those two functions. I'll get on it